code/modules/projectiles/gun.dm

```diff
@@ -144,6 +151,10 @@
 		if(!can_trigger_gun(L))
 			return
 
+	if(safetyon == 1)
+		shoot_with_empty_chamber(user)
```

Comment:
A little suggestion on my side, let's say someone didn't read a changelog and didn't examine the gun before firing.
Is it intended that player doesn't know at first if the gun can't fire because it's out of bullets, or because the safety is on?
I have added a line right after this one to 156, stating that the safety is on.

user << "<span class='warning'>The gun can't fire with safety on!</span>"

Well yeah, it may be stupid to give a hint why the gun won't fire, as it sure is fun to see your confused enemy making a futile attempts to shoot you from your own gun after you got disarmed.
But it can help if someone is trying to reload straight ahead instead of checking the gun, and even that moment of realisation the safety is on when you see that in the chat, takes time to react, enough of time for you to get shot before you can even fire your own gun.
It's up to you to decide if this line is required or not, it's might not be a big difference, yet it is.